projects/swiot1l/src/examples/swiot1l-mqtt/swiot1l_mqtt.c

```diff
@@ -0,0 +1,282 @@
+
+	no_os_gpio_get(&swiot_led1_gpio, &swiot_led1_ip);
+	no_os_gpio_get(&swiot_led2_gpio, &swiot_led2_ip);
+	no_os_gpio_get(&max14906_d1_gpio, &max14906_d1_ip);
+	no_os_gpio_get(&max14906_d2_gpio, &max14906_d2_ip);
+	no_os_gpio_get(&max14906_d3_gpio, &max14906_d3_ip);
+	no_os_gpio_get(&max14906_d4_gpio, &max14906_d4_ip);
+	no_os_gpio_direction_output(max14906_d1_gpio, 0);
+	no_os_gpio_direction_output(max14906_d2_gpio, 0);
+	no_os_gpio_direction_output(max14906_d3_gpio, 0);
+	no_os_gpio_direction_output(max14906_d4_gpio, 0);
+	no_os_gpio_get(&max14906_en_gpio, &max14906_en_ip);
+	no_os_gpio_direction_output(max14906_en_gpio, 0);
+	no_os_gpio_get(&adin1110_cfg0_gpio, &adin1110_cfg0_ip);
+	no_os_gpio_get(&ad74413r_ldac_gpio, &ad74413r_ldac_ip);
+	no_os_gpio_get(&ad74413r_reset_gpio, &ad74413r_reset_ip);
+	no_os_gpio_get(&ad74413r_irq_gpio, &ad74413r_irq_ip);
+	no_os_gpio_get(&max14906_synch_gpio, &max14906_synch_ip);
+	no_os_gpio_get(&adin1110_swpd_gpio, &adin1110_swpd_ip);
+	no_os_gpio_get(&adin1110_tx2p4_gpio, &adin1110_tx2p4_ip);
+	no_os_gpio_get(&adin1110_mssel_gpio, &adin1110_mssel_ip);
+	no_os_gpio_get(&adin1110_cfg1_gpio, &adin1110_cfg1_ip);
+	no_os_gpio_get(&adin1110_int_gpio, &adin1110_int_ip);
+	no_os_gpio_direction_output(ad74413r_ldac_gpio, 0);
+	no_os_gpio_direction_output(ad74413r_reset_gpio, 1);
+	no_os_gpio_direction_output(max14906_synch_gpio, 1);
+	no_os_gpio_direction_output(adin1110_swpd_gpio, 1);
+	no_os_gpio_direction_output(adin1110_tx2p4_gpio, 0);
+	no_os_gpio_direction_output(adin1110_mssel_gpio, 1);
+	no_os_gpio_direction_output(adin1110_cfg1_gpio, 1);
+	no_os_gpio_direction_output(adin1110_cfg0_gpio, 1);
+	no_os_gpio_direction_output(swiot_led1_gpio, 1);
+	no_os_gpio_direction_output(swiot_led2_gpio, 1);
+	no_os_gpio_direction_input(adin1110_int_gpio);
+	no_os_gpio_direction_input(ad74413r_irq_gpio);
+
+	ret = ad74413r_init(&ad74413r, &ad74413r_ip);
+	if (ret)
+		return ret;
+
+	ad74413r_set_channel_function(ad74413r, 0, AD74413R_VOLTAGE_IN);
+	ad74413r_set_channel_function(ad74413r, 1, AD74413R_CURRENT_OUT);
+	ad74413r_set_channel_function(ad74413r, 2, AD74413R_RESISTANCE);
+	ad74413r_set_channel_function(ad74413r, 3, AD74413R_VOLTAGE_OUT);
+
+	ad74413r_set_adc_rejection(ad74413r, 0, AD74413R_REJECTION_NONE);
+	ad74413r_set_adc_rejection(ad74413r, 1, AD74413R_REJECTION_NONE);
+	ad74413r_set_adc_rejection(ad74413r, 2, AD74413R_REJECTION_NONE);
+	ad74413r_set_adc_rejection(ad74413r, 3, AD74413R_REJECTION_NONE);
+
+	ad74413r_set_channel_dac_code(ad74413r, 1, 1000);
+	ad74413r_set_channel_dac_code(ad74413r, 3, 3000);
+
+	memcpy(adin1110_ip.mac_address, adin1110_mac_address, NETIF_MAX_HWADDR_LEN);
+	memcpy(lwip_ip.hwaddr, adin1110_mac_address, NETIF_MAX_HWADDR_LEN);
+
+	ret = no_os_lwip_init(&lwip_desc, &lwip_ip);
+	if (ret)
+		return ret;
+
+	struct tcp_socket_init_param tcp_ip = {
+		.net = &lwip_desc->no_os_net,
+		.max_buff_size = 0
+	};
+
+	ret = socket_init(&tcp_socket, &tcp_ip);
+	if (ret)
+		return ret;
```

Comment:
any memory leaks we should handle?
also, in case of returning, especially if !0, maybe it would be a good idea to print some error messages which would explain what has happened?